index.bs

```diff
@@ -251,21 +251,26 @@ navigator.xr.supportsSessionMode('immersive-vr').then(() => {
 </pre>
 </div>
 
-The {{XR}} object has an <dfn>active immersive session</dfn>, which MUST be initially <code>null</code>, and a <dfn>list of inline sessions</dfn>, which MUST be initially empty.
+The {{XR}} object has a <dfn>pending immersive session</dfn> boolean, which MUST be initially <code>false</code>, an <dfn>active immersive session</dfn>, which MUST be initially <code>null</code>, and a <dfn>list of inline sessions</dfn>, which MUST be initially empty.
 
 <div class="algorithm" data-algorithm="request-session">
 
 When the <dfn method for="XR">requestSession(|options|)</dfn> method is invoked, the user agent MUST return [=a new Promise=] |promise| and run the following steps [=in parallel=]:
 
+  1. Let |mode| be the {{XRSessionCreationOptions/mode}} attribute of the |options| argument.
+  1. Let |immersive| be a boolean set to <code>true</code> if |mode| is {{immersive-vr}} or {{immersive-ar}} and <code>false</code> otherwise.
+  1. If |immersive| is <code>true</code>: 
+    1. If [=pending immersive session=] is <code>true</code> or [=active immersive session=] in not <code>null</code>, [=reject=] |promise| with an {{InvalidStateError}} and abort these steps.
+    1. Set [=pending immersive session=] to be <code>true</code>.
   1. [=Ensure an XR device is selected=].
   1. If [=XR device=] is <code>null</code>, [=reject=] |promise| with <code>null</code>.
-  1. Let |mode| be the {{XRSessionCreationOptions/mode}} attribute of the |options| argument.
-  1. If |mode| is in not in [=XR device=]'s [=list of supported modes=], [=reject=] |promise| with a {{NotSupportedError}} and abort these steps.
-  1. If |mode| is {{immersive-vr}} or {{immersive-ar}} and the [=active immersive session=] is not <code>null</code>, [=reject=] |promise| with an {{InvalidStateError}} and abort these steps.
-  1. If |mode| is {{immersive-vr}} or {{immersive-ar}} and the algorithm is not [=triggered by user activation=], [=reject=] |promise| with a {{SecurityError}} and abort these steps.
+  1. Else if |mode| is in not in [=XR device=]'s [=list of supported modes=], [=reject=] |promise| with a {{NotSupportedError}}.
+  1. Else If |immersive| is <code>true</code> and the algorithm is not [=triggered by user activation=], [=reject=] |promise| with a {{SecurityError}} and abort these steps.
+  1. If |promise| was [=rejected=] and |immersive| is <code>true</code>, set [=pending immersive session=] to be <code>false</code>.
+  1. If |promise| was [=rejected=], abort these steps.
   1. Let |session| be a new {{XRSession}}.
```

Comment:
This is a good question that I don't really know the answer to? I'm gonna push this PR for now and then look into how other specs handle algorithmic steps that boil down to "UA-specific behavior here, if needed", because that seems like it could apply almost anywhere? But I agree that there's benefit to saying "consent prompts should go here."

Reply:
Sounds good.  Can we file an issue against the repo to track this?